src/compiler/program.ts

```diff
@@ -430,13 +430,14 @@ namespace ts {
         return program;
 
         function getCommonSourceDirectory() {
-            if (typeof commonSourceDirectory === "undefined") {
-                if (options.rootDir && checkSourceFilesBelongToPath(files, options.rootDir)) {
+            if (commonSourceDirectory === undefined) {
+                const emittedFiles = filterSourceFilesInDirectory(files, isSourceFileFromExternalLibrary);

```

Comment:
Yes. For the `--outDir` case that makes sense since we are just emitting `.ts` files. Can you provide another case where we would want the `.d.ts` files included?